File: rust/src/y2022/day1.rs
use std::fmt::Display;

pub fn solve(input: &str) -> (Box<dyn Display>, Box<dyn Display>) {
    let elves = input.split("\n\n").map(|s| s.lines().map(|l| l.parse::<u64>().unwrap()));
    let mut totals: Vec<u64> = elves.map(|a| a.sum::<u64>()).collect();
    totals.sort();

    (Box::new(*totals.iter().max().unwrap()), Box::new(totals.iter().rev().take(3).sum::<u64>()))
}

File: clojure/src/y2022/day1.clj
(ns y2022.day1
  (:require
   [clojure.string :refer [split split-lines]]))

(defn solve [input]
  (let [bags (map split-lines (split input #"\n\n"))
        counts (map #(apply + (map read-string %)) bags)
        top3 (take 3 (reverse (sort counts)))
        part1 (apply max counts)
        part2 (apply + top3)]
    [part1 part2]))
